remme/clients/basic.py

```diff
@@ -194,31 +194,41 @@ def _socket_request(self, suffix, data=None):
             raise ClientException('Suffix "%s" not supported' % suffix)
 
     def _handle_response(self, msg_type, resp_proto, req):
-        self._stream.wait_for_ready()
-        future = self._stream.send(
-            message_type=msg_type,
-            content=req.SerializeToString())
+        # Loops while NOT_READY
+        while True:
```

Comment:
What is that?)

Reply:
A way to complete _handle_response when the node is "NOT_READY"

Reply:
How will it break the loop?

Reply:
we have a `break` construct in place for all successful responses or raising exception for failed ones

Reply:
I do not see any `break` here

Reply:
<img width="498" alt="screen shot 2018-08-24 at 10 43 53 am" src="https://user-images.githubusercontent.com/6625496/44571839-a9268700-a78a-11e8-87c3-b8cf738736c9.png">

Reply:
This should be wrapped in an asynchronous way. Our code base is moving to asyncio.